pages/playlist/contributor/[id]/index.vue

```diff
@@ -21,11 +21,16 @@ const onPressPlay = () => {
   }
 };
 const onPressShuffle = async () => {
-  const shuffledTracks = (await useContributorShuffle(contributorId)).data
-    .value;
+  try {
+    const shuffledTracks = (await useContributorShuffle(contributorId)).data
+      .value;
 
-  if (shuffledTracks) {
-    setQueue(shuffledTracks);
+    if (shuffledTracks) {
+      setQueue(shuffledTracks);
+    }
+  } catch (e) {
```

Comment:
I don't like that every api call needs to be wrapped in a try catch. I'd prefer if we detect the error in the middleware and show a message. Then not every component needs to do it itself.

Reply:
That's part of #33. I wouldn't do this in a middleare, because some errors are expected to be handled, others are rather general. I'd rather wrap them all in a wrapper where the caller itself can handle the expected ones and let the wrapper catch all other errors, which should also be reported differently, IMO. That's also why I'm rather against the current implementation of error reporting.